core/primitives/src/shard_layout.rs

```diff
@@ -104,17 +104,19 @@ impl ShardLayout {
     }
 
     #[inline]
-    pub fn get_split_shards(&self, parent_shard_id: ShardId) -> Option<Vec<ShardUId>> {
+    pub fn get_split_shard_uids(&self, parent_shard_id: ShardId) -> Option<Vec<ShardUId>> {
+        self.get_split_shard_ids(parent_shard_id).map(|shards| {
+            shards.into_iter().map(|id| ShardUId::from_shard_id_and_layout(id, &self)).collect()
+        })
+    }
+
+    #[inline]
+    pub fn get_split_shard_ids(&self, parent_shard_id: ShardId) -> Option<Vec<ShardId>> {
```

Comment:
Feels like some methods here don't need to be `#[inline]`.

Reply:
I'll clean those up